lib/rules/no-extra-parens.js

```diff
@@ -263,9 +265,26 @@ module.exports = {
          * @private
          */
         function report(node) {
-            const previousToken = sourceCode.getTokenBefore(node);
-
-            context.report(node, previousToken.loc.start, "Gratuitous parentheses around expression.");
+            const leftParenToken = sourceCode.getTokenBefore(node);
+            const rightParenToken = sourceCode.getTokenAfter(node);
+
+            context.report({
+                node,
+                loc: leftParenToken.loc.start,
+                message: "Gratuitous parentheses around expression.",
+                fix(fixer) {
+                    const parenthesizedSource = sourceCode.text.slice(leftParenToken.range[1], rightParenToken.range[0]);
+                    const precedingToken = sourceCode.getTokenBefore(leftParenToken);
+
+                    // If the parens are preceded by a keyword (e.g. `typeof(0)`), a space should be inserted (`typeof 0`)
+                    const isPrecededByKeyword = precedingToken && precedingToken.type === "Keyword" && precedingToken.end === leftParenToken.start;
```

Comment:
That fix works, but I wonder if it would be better to just always output a space after keywords, and then let the [keyword-spacing](http://eslint.org/docs/rules/keyword-spacing) rule clean up if the spacing is wrong.

On the other hand, I suppose it's better to modify the code as little as possible.